spyder_unittest/unittestplugin.py

```diff
@@ -24,16 +26,67 @@
 _ = get_translation("unittest", dirname="spyder_unittest")
 
 
-class UnitTestPlugin(UnitTestWidget, SpyderPluginMixin):
-    """Unit testing."""
+class UnitTestWidgetInSpyder(UnitTestWidget):
+    """
+    Unit test widget for use inside Spyder.
+
+    This class overrides some functions in `UnitTestWidget` to provide better
+    integration with Spyder.
+    """
+
+    def __init__(self, main):
+        """
+        Constructor.
+
+        Parameters
+        ----------
+        main : MainWindow
+            Spyder main window
+        """
+        UnitTestWidget.__init__(self, parent=main)
```

Comment:
The problem is that I don't understand how the widget is supposed to get the return value.

In this case, I want to get the file name of the current active file in the editor in order to provide a sensible default when asking the user for the directory in which to run the tests in.